### DISCUSSION TITLE
Understanding of the deformable body physics

### DISCUSSION THREAD

**Discussion by Servo97** [google-deepmind/mujoco#1740] [General]
Hi,
I'm a grad student trying to model deformable objects in mujoco using flex. I have a few questions about friction in flex bodies. 
I will use the following code and the attached video to ground my discussion:

```
<mujoco model="handkerchief">
    <extension>
        <plugin plugin="mujoco.elasticity.solid" />
    </extension>
    <worldbody>
        <body name="v1" pos="0 0 0.005">
            <inertial pos="0 0 0" mass="0.01" diaginertia="1.66667e-05 1.66667e-05 1.66667e-05" />
        </body>
        <body name="v2" pos="0.1 0 0.005">
            <inertial pos="0 0 0" mass="0.01" diaginertia="1.66667e-05 1.66667e-05 1.66667e-05" />
            <freejoint />
        </body>
        <body name="v3" pos="0 0.1 0.005">
            <inertial pos="0 0 0" mass="0.01" diaginertia="1.66667e-05 1.66667e-05 1.66667e-05" />
            <freejoint />
        </body>
        <body name="v4" pos="0.1 0.1 0.005">
            <inertial pos="0 0 0" mass="0.01" diaginertia="1.66667e-05 1.66667e-05 1.66667e-05" />
            <freejoint />
        </body>
    </worldbody>
    <deformable>
        <flex name="FL1" dim="2" body="v1 v2 v3" vertex="0 0 0  0 0 0  0 0 0" element="0 1 2" rgba="1 0 0 0.9" />
        <flex name="FL2" dim="2" body="v2 v3 v4" vertex="0 0 0  0 0 0  0 0 0" element="2 1 0" rgba="0 1 0 0.9" />
    </deformable>
    <equality>
        <flex flex="FL1" />
        <flex flex="FL2" />
    </equality>
</mujoco>
```

So iiuc, the bodies we define to be used in <deformable> tags, have some mass and will fall down if initialized in the air with a freejoint. However, they don't seem to obey friction laws unless I attach small \< geom \>s to the bodies that can take care of friction. 

Is there a way to define some friction properties so that purely deformable bodies can interact with the ground? e.g. stop rotating on the plane after applying some force by mouse-dragging as can not be seen from the video below?

TLDR: Flex bodies keep rolling on the ground for infinity once an initial force is applied. Is there a way to make them obey friction without adding tiny \< geom \> objects to the vertices?

https://github.com/google-deepmind/mujoco/assets/22110708/84dbd788-9b17-4438-b248-0e8395e0126e

**Comment by quagla** [comment]
Hi, you are right and there is currently no way of doing that as far as I know..

**Reply by Servo97** [reply]
I see! 
I'm trying to see if we can simulate origami robots in mujoco using flex. The structures do look promising with a bit of a learning curve. 

Do you have any recommendations on how we can apply friction as elegantly as possible? We are thinking both locomotion and manipulation origami structures. That would imply friction on a random <freejoint/> body as well as on the ground. 

The best solution I have is to attach the small geom elements as I mentioned before 😅

Thanks!!
